internal/ethapi/transaction_args.go

```diff
@@ -177,6 +177,20 @@ func (args *TransactionArgs) setDefaults(ctx context.Context, b Backend) error {
 
 // setFeeDefaults fills in default fee values for unspecified tx fields.
 func (args *TransactionArgs) setFeeDefaults(ctx context.Context, b Backend) error {
+	head := b.CurrentHeader()
+	// Sanity check the EIP-4844 fee parameters.
+	if args.BlobFeeCap != nil && args.BlobFeeCap.ToInt().Sign() == 0 {
+		return errors.New("maxFeePerBlobGas must be non-zero")
+	}
+	if b.ChainConfig().IsCancun(head.Number, head.Time) {
```

Comment:
This part I don't agree with. Geth should always be able to *create* blob transactions, it shouldn't depend on the chain head for that. If there is a restriction about sending them, it will be enforced by the txpool, but AFAIK even the pool accepts them before Cancun.

Reply:
Sure. I liked to have it because setDefaults doubles as a pre-validation for the EVM sometimes (createAccessList). Removed that check.